Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift

```diff
@@ -360,7 +368,7 @@ extension DefaultDiagnosticConsoleFormatter {
             // Example:
             // 9  | A line outside the diagnostic range.
             // 10 + A line inside the diagnostic range.
-            result.append("\n\(linePrefix) \(separator) \(highlightedSource)")
+            result.append("\n\(linePrefix) \(separator) \(highlightedSource)".removingTrailingWhitespace())
```

Comment:
Yes. I was updating a test diagnostics formatter [here](https://github.com/apple/swift-docc/pull/756/files#diff-856201a2266867cc3bce33c8e301d89e125a957bd4f7e0d834abd45f5fd4c561R249) to use the test file system and I wanted to assert that the diagnostic message was logged. 

The reason for this change is that the two empty lines of source in the diagnostic message in that test was really annoying to work with because the console writer didn't trim the trailing whitespace, so it printed those two lines as `2 | ` and `4 | ` (with a trailing space). The issue is that Xcode automatically trims trailing whitespace while editing so I couldn't write the string literal for the expected diagnostic message because Xcode would remove the trailing whitespace and make the test fail.

I could disable that setting in Xcode, write the string literal, and then enable that setting again, but anyone else who touched that test in the future would break it. 

I could also write the test so that it splits the diagnostic message by line, trims the trailing whitespace, and joins the diagnostic message again before comparing it to the expected value, but we would have to repeat that code in every test that wanted to assert that a diagnostic message was printed as expected.

So instead I decided to trim the trailing whitespace in the diagnostic message so that it will compare nicely to the expected output in this test and in future tests.